% Chapter X

\chapter{Perspectives of genomics for infection biology} % Chapter title

\label{ch:03-02} % For referencing the chapter elsewhere, use \autoref{ch:name} 

%----------------------------------------------------------------------------------------

\section{The 3D genome and the advent of deep learning}
% Deepnog (annotation)
% DeepHiC, SRHic, ...(resolution)
% akita & deepc
It can look attractive to produce a model of the 3D genome, ideally a predictive one, that would be able to infer how the structure reacts to specific changes. However, in many organisms, the rules governing genome organization are intricate and it would be unwieldy to model them explicitly. Deep learning provides an attractive framework to produce such a model without knowing all the rules involved. There are already successful applications of deep learning in biology for various different tasks such as gene annotation \cite{stiehlerHelixerCrossspeciesGene2020,khodabandelouGenomeAnnotationSpecies2020}, variant calling \cite{poplinUniversalSNPSmallindel2018}, classification of coding RNA \cite{hillDeepRecurrentNeural2018}, prediction of nucleosome positioning \cite{routhierGenomewidePredictionDNA2021} and perhaps most importantly, protein structure prediction \cite{jumperHighlyAccurateProtein2021}. More recently progress has also been made for the prediction prediction gene expression and promoter-enhancer interactions solely from the DNA sequence \cite{avsecEffectiveGeneExpression2021}.

%cite akita and deep3C
Generally, applications of deep learning in the 3D genome field have been limited to denoising or improving the resolution of Hi-C matrices. Recently however, there have been successful attempts at predicting the structure of mammalian genomes from the DNA sequence, including (and most importantly) the prediction of conformational changes induced by mutations \cite{fudenbergPredicting3DGenome2020,schwessingerDeepCPredicting3D2020}. In the future, these approaches could be helpful to identify mutations or regions to focus on, and one could imagine it being used to model the consequences of infection on the host.

Unfortunately, several limitations must be overcome before deep learning methods can become an amenable tool to understand the relationship between biological processes and genome structure. First, it requires tremendous amounts of training data, which in the case of Hi-C remains expensive to generate. Then, such model would also require information about all the factors at play in the process, which we do not know. Even in the event that we manage to obtain a model that effectively predicts conformational changes, in most cases this is still unsatisfying: The general scientific interest is usually to understand the rules and logic that connect the biological process (e.g. infection) to structural changes. In deep learning models these rules are obscured, taking the form of large weight matrices, and extracting biological meaning from them would require consequent advances in model interpretability \cite{talukderInterpretationDeepLearning2021}.
