% Chapter X

\chapter{Biological and technical discussions} % Chapter title
\label{ch:03-01} % For referencing the chapter elsewhere, use \autoref{ch:name} 

\section{Plasticity of intracellular bacteria}

Throughout the previous part, we have developed new methods to approach the detection and quantification of chromatin features and their changes during infection (Chap. \ref{ch:02-01}). We then applied these methods on two different infection settings: Infection of the amoeba \textit{A. castellanii} by \textit{L. pneumophila} (Chap. \ref{ch:02-02}) and of murine bone marrow macrophages by \textit{S. enterica} (Chap. \ref{ch:02-03}). Although both are intracellular bacteria with similar infection strategies, they can infect completely different hosts. The size of the mouse haploid genome outclasses that of \textit{A. castellanii} by two orders of magnitude (4.3Gbp vs 45Mbp) and its spatial organization is much more complex, with compartments and intricate nested loops bridging very long distances. Despite all their differences, both unicellular and human hosts are susceptible to \textit{L. pneumophila} infection. This is most impressive knowing that human is an evolutionary dead-end for \textit{L. pneumophila} due to the absence of human-to-human transmission. The bacterial genome is therefore shaped exclusively by selective pressure in its unicellular hosts. The ability to infect multicellular hosts is probably linked to the high conservation of targeted pathways and has been attributed to the wide range of protozoan hosts infected by the bacterium \cite{molofskyDifferentiateThriveLessons2004}.

This conservation was also visible in our results, as several redundant processes are deregulated during infection \textit{Legionella} and \textit{Salmonella}, such as cell cycle regulation, protein ubiquitination and transmembrane transport.


%----------------------------------------------------------------------------------------

\section{Combination of effects}

The analyses presented in this work focus on the description of changes happening in the 3D genome during infection. One issue with this type of experiments is that we observe the combined effect of the pathogen activity and the host immune response. There are means to dampen one of these effects, such as the use of mutant pathogens unable secrete effector proteins as control to trigger host response (as used in Chapter \ref{ch:02-03}). However, these controls do not completely emulate the pathogen activity, as they will not replicate and generally \cite{vogelConjugativeTransferVirulence1998} and therefore will not elicit the same immune response. Decoupling and deciphering the different factors at play during infection ultimately requires the use of mutagenesis screen using Transposon insertion or CRISPR.

Such approaches, however, are not yet applicable to study the 3D genome, mostly due to the cost and time of Hi-C related techniques. Regardless, more descriptive approaches such as the ones used in this work are still important to understand the extent of changes happening during infection. Specifically, they can still inform us on the type of structural changes that the genome undergoes during infection and the global importance of genome organization during infection.


\section{Reproducibility and reliability challenges}
% Impact of the reproducibility crisis
Results from genomics analyses are especially sensitive to the parameters and methods used. This makes reproducibility in bioinformatics of utmost importance. Just like RNAseq, Hi-C has important technical variability which needs to be accounted for using multiple replicates.

It was proposed that RNAseq experiments for differential expression analysis should comprise at least 6 replicates and ideally 12 \cite{schurchHowManyBiological2016}. While this is probably true for most omics experiments, this entails a high cost which is often the limiting factor when designing experiments in genomics.

The core issue with low replicate numbers is the impossibility to distinguish between technical variability due to the assay and biological variability of interest. As a consequence, when fewer replicates are used, lower effect size (fold changes in the case of gene expression) become undetectable. This is especially problematic when studying gene regulation, where small changes in expression can be important.

% Mention beyond p-values ?

% Software quality
Unlike RNAseq, where the standard for analyses is well established and most softwares are able to account for replicates and experimental design, most softwares available for Hi-C analysis cannot use replicate information. This limits the power of analysis to detect only very strong changes. The lack of standard also causes a general fragmentation of bioinformatic tools, with many redundant tools of variable quality. One recurrent issue is the absence or low quality of unit tests and documentation, which are unfortunately still not regarded as standard in the computational biology community. Unit tests could be viewed as an equivalent to control experiments in molecular biology, as they validate each logic piece of the software using inputs with known truths. Software lacking these controls is more likely to have undetected bugs that could impact results and lead to false conclusion.
% The importance of standardisation
Another issue with newer techniques like Hi-C is the lack of standardisation for file formats and practices. This can result in incompatibilities between programs and introduce errors during conversions that alter the data.

Some general practices can of course be adopted to address these issues, such as writing comprehensive documentation, solid tests and maintain software, but as it stands, there is no incentive to do so in academia. Ultimately, these directives need to be enforced globally in the peer review process by journals, so that tools need to meet quality standards to reach publication.

Although this will increase the effort and time required to develop methods, this also means the resulting tools will be more reliable, easier to use and more widely adopted. Fortunately, recent years have seen an increasing adoption of good practices in bioinformatic software and the cool format is now supported by the majority of Hi-C analysis tools. This could mean that the quality of academic software for bioinformatics will undergo major improvements in the forseeable future. 