% Chapter X

\chapter{Biological and technical discussions} % Chapter title
\label{ch:03-01} % For referencing the chapter elsewhere, use \autoref{ch:name} 

\section{Representation of protozoan genomes}

The representation of species with sequenced genomes is traditionally skewed towards mammals and vertebrate animals. Many groups which are much more abundant in nature or ecologically important are underrepresented in reference genome databases \cite{davidSequencingDisparityGenomic2019}. \textit{Acanthamoeba} provides a fitting example, as they are ubiquitous in aquatic environment and form important interactions with numerous other microorganisms, yet have no chromosome quality reference available. The reference genomes of the two \textit{A. castellanii} strains generated here provide the first high quality reference in the \textit{Acanthamoeba} group and thus a valuable resource for the comparative study of amoebae.

\section{Host plasticity of intracellular bacteria}

Throughout the previous part, we have developed new approaches to detect chromatin features and quantify their changes during infection (Chap. \ref{ch:02-01}). We then applied these methods in two different infection settings: Infection of the amoeba \textit{A. castellanii} by \textit{L. pneumophila} (Chap. \ref{ch:02-02}) and of murine bone marrow macrophages by \textit{S. enterica} (Chap. \ref{ch:02-03}). Although both are intracellular bacteria with similar infection strategies, they can infect completely different hosts. The size of the mouse haploid genome outclasses that of \textit{A. castellanii} by two orders of magnitude (4.3Gbp vs 45Mbp) and its spatial organization is much more complex, with A/B compartmentation and intricate nested loops bridging very long distances. Despite all their differences, both unicellular and human hosts are susceptible to \textit{L. pneumophila} infection. This is most impressive knowing that human is an evolutionary dead-end for \textit{L. pneumophila} due to the absence of human-to-human transmission. The bacterial genome is therefore shaped exclusively by selective pressure in its unicellular hosts. The ability to infect multicellular hosts is probably linked to the high conservation of targeted pathways and has been attributed to the wide range of protozoan hosts infected by the bacterium \cite{molofskyDifferentiateThriveLessons2004}.

This conservation was also visible in our results, as several processes deregulated during infection are common between \textit{Legionella} and \textit{Salmonella}, such as cell cycle regulation, protein ubiquitination and transmembrane transport.

%----------------------------------------------------------------------------------------

\section{Combination of effects}

The analyses presented in this work focus on the description of changes happening in the 3D genome during infection. One issue with this type of experiments is that we observe the combined effect of the pathogen activity and the host immune response. There are means to dampen one of these effects, such as the use of mutant pathogens which are unable to secrete effector proteins as control to trigger host response (as used in Chapter \ref{ch:02-03}). Although these controls do not completely emulate the pathogen activity, as they will not replicate  \cite{vogelConjugativeTransferVirulence1998} and therefore will not elicit the same immune response, they are still useful to separate the effect of the infection. In the case of \textit{L. pneumophila} (Chap. \ref{ch:02-02}), reproducing the infections with mutants for the dotA secretion system and romA methyl-transferase would allow to further isolate the chromatin changes due to the infection and romA activity.

At large scale, decoupling and deciphering the different factors at play during infection ultimately requires the use of mutagenesis screen using Transposon insertion or CRISPR. Such approaches, only assess the effect on host survival and not chromatin changes, and to our knowledge there is no method to screen for chromatin modifiers. Regardless, more descriptive approaches such as the ones used in this work are still important to understand the extent of changes happening during infection. Specifically, they can still inform us on the type of structural changes that the genome undergoes and global importance of genome organization during infection.


\section{Reproducibility and reliability challenges}

% Impact of the reproducibility crisis
Results from genomics analyses are especially sensitive to the parameters and methods used. This makes reproducibility in bioinformatics of utmost importance. Just like RNA-seq, Hi-C also has important technical variability which needs to be accounted for using multiple replicates.

It was proposed that RNA-seq experiments for differential expression analysis should comprise at least 6 replicates and ideally 12 \cite{schurchHowManyBiological2016}. While this is probably true for most omics experiments, this entails a high cost which is often the limiting factor when designing experiments in genomics.

The core issue with low replicate numbers is the lack of power to distinguish between biological variability across replicates and differences due to the condition of interest. As a consequence, when fewer replicates are used, lower effect sizes (fold changes in the case of gene expression) become undetectable. This is especially problematic when studying gene regulation, where small changes in expression could be relevant.

% Mention beyond p-values ?

% Importance of software quality and standardisation
Unlike RNA-seq, where the standard for analyses is well established and most softwares are able to account for replicates and experimental design, most methods available for Hi-C analysis do not leverage replicate information. This limits the power of analysis to detect only major changes. The lack of standards for Hi-C data formats and processing also causes a general fragmentation of bioinformatic tools, with many redundant softwares of variable quality. One recurrent issue is the absence or low quality of unit tests and documentation, which are unfortunately still not regarded as standard in the computational biology community. Unit tests could be viewed as an equivalent to control experiments in molecular biology, as they validate each logic piece of the software using inputs with known truths. Software lacking these controls is more likely to have undetected bugs that could impact results and lead to false conclusions.
% The importance of standardisation

Some general practices can of course be adopted to address these issues, such as writing comprehensive documentation, solid tests and maintain software, but as it stands, there is no incentive to do so in academia. Ultimately, some of these directives need to be enforced globally in the publishing process to ensure tools meet quality standards to reach publication.

Although adopting such practices will increase the effort and time required to develop methods, this also means the resulting tools will be more reliable, easier to use and more widely adopted. Fortunately, recent years have seen an increasing adoption of good practices in bioinformatic software. One such example is the \textit{nf-core} ecosystem backed by SciLifeLab, a public institution dedicated to open-source scientific software development \url{https://www.scilifelab.se}. The generalization of similar initiatives could mean that the quality of academic software for bioinformatics will undergo major improvements in the forseeable future \cite{ewelsNfcoreFrameworkCommunitycurated2020,wrattenReproducibleScalableShareable2021}.