% Chapter X

\chapter{Infection through the lense of genomics} % Chapter title

\label{ch:01-02} % For referencing the chapter elsewhere, use \autoref{ch:name} 

%----------------------------------------------------------------------------------------

The toolset to detect and investigate bacterial infection traditionally included biochemical assays and microscopy. The recent technological advances in DNA sequencing have spurred a rapid extension of this toolset with NGS-derived methods. Here we introduce the different ways genomics can provide biological insights into the biology of bacterial pathogens.

\section{Pathogen characterization}

The most fundamental task related to infection in biomedical research is to detect the presence of infectious agents and characterize them. This allows to test patients presenting suspicious symptoms for the presence of known pathogens, or determine the pathogenicity of a particular strain. 

Genotyping was traditionally achieved using molecular biology techniques, such as \acrfull{RFLP} or \acrfull{PFGE}. These techniques rely on the negative charge of the DNA molecules. When put in a gel submitted to an electromagnetic field, these molecules will migrate along the electrical current. The migration distance is proportional to the size of molecules. After migration is complete, the gel can be treated with chemical markers to highlight the location of DNA molecules. This will reveal discrete bands of similar-length DNA fragments. Together these bands form a bar-code which can be interpreted by the scientist to draw conclusion about the number and size of these fragments. In the case of \acrshort{RFLP}, the genome is prealably digested by \Gls{restriction enzyme}s. The digestion will result in a series of fragments whose lengths can be seen on the gel. Bacterial genotypes wil have different mutations which will affect the digestion pattern and resulting barcode on the gel.

While these methods work well to determine differences between alleles, they do not inform us on the actual DNA sequence involved. The advent of DNA sequencing made it possible to directly link phenotype with associated sequences of nucleotides. \acrfull{WGS} provides accurate information on an organism's genotype, down to down to the \acrfull{SNP}, allowing to define genotypes at a finer scale. The main shortcoming of \acrshort{WGS} is its higher cost than other genotyping techniques, but the recent plummeting of sequencing costs have made it releatively affordable. These advantages have made \acrshort{WGS} a popular approach in clinical settings.

\section{Genomics to probe homeostasis}

When host cells are exposed to or infected by a pathogen, their homeostatic state will be disrupted. This disruption is a combiantion of host-triggered reactions to improve its survival, and alterations caused by the pathogen to colonize the host cell. Untangling these two phenomenon is a complex issue in itself, but investigating what biological functions or pathways are disrupted upon infection can already give good  pointers to important players in the infeciton.

Multiple levels of regulation are affected, from gene expression to epigenetic states, and over the years, a vast arsenal of NGS techniques have been developed to read thse regulatory states. For example, ChIPseq, ...

he transcribed genes, present in the cell in the form of RNA, can also be read using the same DNA sequencing technologies. This allows one to estimate the expression of genes from the amount of RNA present in the cells. This is useful when studying infection, as the expression of each gene can be compared between uninfected and infected cells to detect which biological functions are perturbed.



\section{Capturing chromosome conformation}

The use of genomics to investigate the three-dimensional organisation of the genome started with the invention of \acrfull{3C} \cite{Dekker2002}. This technique allowed to measure the frequency of physical interactions a pair of loci. This is done by crosslinking the genome with formaldehyde, which forms stable bonds between DNA and proteins, and subsequently digesting the genome with a restriction enzyme. The digested genome is then relilgated, and the religation will happen with different neighbouring fragment. Loci which are closer in space will be religated more often with each other in the population of cells. The crosslink is then reverted and qPCR is used to measure the quantity of religated products containing the two loci of interest. 

Since then, many derivative of the \acrshort{3C} technique have been developed. The most significant improvement was brought by Hi-C. This method works similarly to 3C except that next generation sequencing is used instead of qPCR. This allows to quantify the interaction frequency of all versus all loci in the genome instead of using specific primers for a pair of locus. Hi-C also has an additional step where biotinylated bases are added during religation. This allows to pull-down religation products using streptavidin so that only products that underwent the digestion-religation process are sequenced.

Hi-C allows to generate interaction frequency maps of the whole genome wich reflect its 3D structure.

\section{Combining layers of biological informations}

\section{Reproducibility and reliability challenges}
% Impact of the reproducibility crisis
% Software quality
% The importance of standardisation
