% Chapter X

\chapter{The importance of genome assembly} % Chapter title

\label{ch:01-03} % For referencing the chapter elsewhere, use \autoref{ch:name} 

%----------------------------------------------------------------------------------------

\section{From contigs to chromosomes}
% Define terminology, explain advantages of chromosome level assemblies and how to generate them
% Subsections for bionano, linked reads and Hi-C
% Maybe a quick mention of haplotype-resolved assemblies and genome-graphs

The emergence of specialized technologies aimed at scaffolding now allows to generate more continuous and correct genomes at reduced costs. One example is the recent rebirth of optical mapping through Bionano's sapphire technology which uses nickases to introduce fluorescent probes into chromosomes at specific sites. The order of these probes and their relative distance form barcodes which can then be used to scaffold genome assemblies, reorder and merge contigs. This is often combined with Hi-C to generate highly continuous assemblies even in the presence of repeated sequences.


\section{Phylogenetic representation}
% HGT detection requires a reference group

\section{The transition to genome graphs}

Until recently, all genome references were stored as linear sequences of DNA. This linear sequence is often obtained from a mix of multiple individuals, or alleles within an individual. It is effectively a semi-arbitrary combination of multiple haplotypes merged into an artificial consensus sequence. A recent alternative is to produce a sequence graph instead. Given a collection of haplotypes, individuals, or strains of a species, one can generate a graph where identical regions are collapsed, while sample specific variants form bubbles retaining the genetic variability. As this approach is very recent, the support for genome graphs is currently limited to a few tools, making their applications very limited.
